dockerfiles/cli/scripts/openshift/deploy_che.sh

```diff
@@ -0,0 +1,319 @@
+while [[ $# -gt 1 ]]
+do
+key="$1"
+case $key in
+    -c | --command)
+    COMMAND="$2"
+    shift
+    ;;
+    *)
+            # unknown option
+    ;;
+esac
+shift
+done
+
+# --------------------------------------------------------
+# Set configuration common to both minishift and openshift
+# --------------------------------------------------------
+
+DEFAULT_COMMAND="deploy"
+COMMAND=${COMMAND:-${DEFAULT_COMMAND}}
+DEFAULT_CHE_IMAGE_REPO="docker.io/eclipse/che-server"
+CHE_IMAGE_REPO=${CHE_IMAGE_REPO:-${DEFAULT_CHE_IMAGE_REPO}}
+DEFAULT_CHE_IMAGE_TAG="nightly-centos"
+CHE_IMAGE_TAG=${CHE_IMAGE_TAG:-${DEFAULT_CHE_IMAGE_TAG}}
+DEFAULT_CHE_LOG_LEVEL="INFO"
+CHE_LOG_LEVEL=${CHE_LOG_LEVEL:-${DEFAULT_CHE_LOG_LEVEL}}
+
+# Keycloak production endpoints are used by default
+DEFAULT_KEYCLOAK_OSO_ENDPOINT="https://sso.openshift.io/auth/realms/fabric8/broker/openshift-v3/token"
+KEYCLOAK_OSO_ENDPOINT=${KEYCLOAK_OSO_ENDPOINT:-${DEFAULT_KEYCLOAK_OSO_ENDPOINT}}
+DEFAULT_KEYCLOAK_GITHUB_ENDPOINT="https://sso.openshift.io/auth/realms/fabric8/broker/github/token"
+KEYCLOAK_GITHUB_ENDPOINT=${KEYCLOAK_GITHUB_ENDPOINT:-${DEFAULT_KEYCLOAK_GITHUB_ENDPOINT}}
+
+# OPENSHIFT_FLAVOR can be minishift or openshift
+# TODO Set flavour via a parameter
+DEFAULT_OPENSHIFT_FLAVOR=minishift
+OPENSHIFT_FLAVOR=${OPENSHIFT_FLAVOR:-${DEFAULT_OPENSHIFT_FLAVOR}}
+
+
+if [ "${OPENSHIFT_FLAVOR}" == "minishift" ]; then
+  # ---------------------------
+  # Set minishift configuration
+  # ---------------------------
+  echo -n "[CHE] Checking if minishift is running..."
+  minishift status | grep -q "Running" ||(echo "Minishift is not running. Aborting"; exit 1)
+  echo "done!"  
+
+  DEFAULT_OPENSHIFT_ENDPOINT="https://$(minishift ip):8443/"
+  OPENSHIFT_ENDPOINT=${OPENSHIFT_ENDPOINT:-${DEFAULT_OPENSHIFT_ENDPOINT}}
+  DEFAULT_OPENSHIFT_USERNAME="developer"
+  OPENSHIFT_USERNAME=${OPENSHIFT_USERNAME:-${DEFAULT_OPENSHIFT_USERNAME}}
+  DEFAULT_OPENSHIFT_PASSWORD="developer"
+  OPENSHIFT_PASSWORD=${OPENSHIFT_PASSWORD:-${DEFAULT_OPENSHIFT_PASSWORD}}
+  DEFAULT_CHE_OPENSHIFT_PROJECT="eclipse-che"
+  CHE_OPENSHIFT_PROJECT=${CHE_OPENSHIFT_PROJECT:-${DEFAULT_CHE_OPENSHIFT_PROJECT}}
+  DEFAULT_OPENSHIFT_NAMESPACE_URL="${CHE_OPENSHIFT_PROJECT}.$(minishift ip).nip.io"
+  OPENSHIFT_NAMESPACE_URL=${OPENSHIFT_NAMESPACE_URL:-${DEFAULT_OPENSHIFT_NAMESPACE_URL}}
+  DEFAULT_CHE_KEYCLOAK_DISABLED="true"
+  CHE_KEYCLOAK_DISABLED=${CHE_KEYCLOAK_DISABLED:-${DEFAULT_CHE_KEYCLOAK_DISABLED}}
+  DEFAULT_CHE_DEBUGGING_ENABLED="true"
+  CHE_DEBUGGING_ENABLED=${CHE_DEBUGGING_ENABLED:-${DEFAULT_CHE_DEBUGGING_ENABLED}}
+
+elif [ "${OPENSHIFT_FLAVOR}" == "osio" ]; then
+  # ----------------------
+  # Set osio configuration
+  # ----------------------
+  if [ -z "${OPENSHIFT_TOKEN+x}" ]; then echo "[CHE] **ERROR** Env var OPENSHIFT_TOKEN is unset. You need to set it with you OSO token to continue. To retrieve your token: https://console.starter-us-east-2.openshift.com/console/command-line. Aborting"; exit 1; fi
```

Comment:
Maybe it should be "with your" instead of "with you"

Reply:
Fixed. Thanks @AndrienkoAleksandr!